src/geode/mesh/core/polygonal_surface.cpp

```diff
@@ -121,11 +122,81 @@ namespace geode
             polygon_around_vertex_->value( vertex_id ) = polygon_vertex;
         }
 
+        index_t find_edge( const std::vector< index_t >& edge_vertices ) const
+        {
+            auto itr = edge_indices_.find( edge_vertices );
+            if( itr != edge_indices_.end() )
+            {
+                return edge_indices_.at( edge_vertices );
+            }
+            return NO_ID;
+        }
+
+        index_t find_or_create_edge(
+            const std::array< index_t, 2 >& edge_vertices )
+        {
+            auto size = edge_indices_.size();
+            std::vector< index_t > vertices( 2 );
+            vertices[0] = edge_vertices[0];
+            vertices[1] = edge_vertices[1];
+            auto id = find_edge( vertices );
+            if( id != NO_ID )
+            {
+                return id;
+            }
+            edge_indices_[vertices] = size;
+            edge_attribute_manager_.resize( size + 1 );
+            return size;
+        }
+
+        void update_edge_vertices( const std::vector< index_t >& old2new )
+        {
+            auto old_edge_indices = std::move( edge_indices_ );
+            edge_indices_.clear();

```

Comment:
the use of a variable after a move is undefined behavior. Don't do this! You are lucky it works.